lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -245,13 +224,16 @@ bool FrontendInputsAndOutputs::verifyInputs(DiagnosticEngine &diags,
 
 void FrontendInputsAndOutputs::clearInputs() {
   AllInputs.clear();
+  InputsByName.clear();
   PrimaryInputs.clear();
 }
 
 void FrontendInputsAndOutputs::addInput(const InputFile &input) {
-  if (!input.file().empty() && input.isPrimary())
-    PrimaryInputs.insert(std::make_pair(input.file(), AllInputs.size()));
+  const unsigned index = AllInputs.size();
   AllInputs.push_back(input);
+  InputsByName.insert(std::make_pair(AllInputs.back().file(), index));
```

Comment:
Do we need to store a mapping for non-primary inputs?

Reply:
Yes, I think so. For instance in WMO mode... But I can try the experiment if you like.

Reply:
Yeah. I can't think of a time when we'd look an input up by name in WMO mode, but maybe I'm just forgetting a case.

Reply:
Did you find the case where this is necessary?